// @devlikeapro/n8n-nodes-chatwoot/dist/credentials/ChatWootApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatWootApi = void 0;
class ChatWootApi {
    constructor() {
        this.name = 'chatwootApi';
        this.displayName = 'ChatWoot API';
        this.documentationUrl = 'https://www.chatwoot.com/docs/contributing-guide/chatwoot-apis';
        this.properties = [
            {
                displayName: 'ChatWoot API URL',
                name: 'url',
                placeholder: "https://www.chatwoot.com",
                type: 'string',
                default: '',
                required: true,
            },
            {
                displayName: "There's many " +
                    "<a href='https://www.chatwoot.com/docs/contributing-guide/chatwoot-apis' target='_blank'></a><b>Access Token Types</b> in <b>ChatWoot</b></a>," +
                    " so we just check that <code>{{url}}/api</code> is accessible without token.",
                default: '',
                name: 'operation',
                type: 'notice',
            },
            {
                displayName: 'Access Token',
                name: 'accessToken',
                type: 'string',
                placeholder: "00000000-0000-0000-0000-000000000000",
                default: '',
                required: true,
                typeOptions: { password: true },
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                headers: {
                    'Api-Access-Token': '={{$credentials.accessToken}}',
                },
            },
        };
        this.test = {
            request: {
                baseURL: '={{$credentials.url}}',
                url: '/api',
            },
        };
    }
}
exports.ChatWootApi = ChatWootApi;
