// @devlikeapro/n8n-nodes-chatwoot/dist/nodes/ChatWoot/ChatWoot.node.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatWoot = void 0;
const n8n_openapi_node_1 = require("@devlikeapro/n8n-openapi-node");
const doc = __importStar(require("./openapi.json"));
const config = {};
const parser = new n8n_openapi_node_1.N8NPropertiesBuilder(doc, config);
const properties = parser.build();
class ChatWoot {
    constructor() {
        this.description = {
            displayName: 'ChatWoot',
            name: 'chatWoot',
            icon: 'file:chatwoot.svg',
            group: ['transform'],
            version: 1,
            subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
            description: 'Interact with ChatWoot API',
            defaults: {
                name: 'ChatWoot',
            },
            inputs: ['main'],
            outputs: ['main'],
            credentials: [
                {
                    name: 'chatwootApi',
                    required: true,
                },
            ],
            requestDefaults: {
                headers: {
                    Accept: 'application/json',
                    'Content-Type': 'application/json',
                },
                baseURL: '={{$credentials.url}}',
            },
            properties: properties,
        };
    }
}
exports.ChatWoot = ChatWoot;
